libcontainer/configs/mount_linux.go

```diff
@@ -29,8 +31,21 @@ type Mount struct {
 
 	// Extensions are additional flags that are specific to runc.
 	Extensions int `json:"extensions"`
+
+	// UID/GID mappings used for changing file owners w/o calling chown, fs should support it.
+	// Every mount point could have its own mapping.
+	UIDMappings []IDMap `json:"uidMappings"`
+	GIDMappings []IDMap `json:"gidMappings"`
 }
 
 func (m *Mount) IsBind() bool {
 	return m.Flags&unix.MS_BIND != 0
 }
+
+func (m *Mount) IsBindOnly() bool {
+	return m.IsBind() && !m.IsIDMap()
+}
+
+func (m *Mount) IsIDMap() bool {
+	return m.IsBind() && (len(m.UIDMappings) > 0 || len(m.GIDMappings) > 0)
```

Comment:
@thaJeztah The only way to create an idmap mount is creating a bind-mount. You can't just mount anything with ID translation, you can only do a bind-mount and add ID mappings to it.

So, do you think is ok if I only rename the function name?

Reply:
That's not entirely correct. :) In the new mount api there's no distinction between an fd gotten from `open_tree(OPEN_TREE_CLONE)` and one gotten from `fsmount()`. Both are `O_PATH` fds that refer to an anonymous mount or mount tree. So:
```
fd_mnt = open_tree("/mnt", OPEN_TREE_CLONE);
mount_setattr(fd_mnt, ...);
```
and

```
fd_fs = fsopen("tmpfs);
fsconfig(FSCONFIG_CMD_CREATE);
fd_mnt = fsmount(fd_fs);
mount_setattr(fd_mnt, ...);
```

both work.

Reply:
@brauner ohh, interesting, thanks! We definitely don't support this in the code, though (we are using open_tree() and just that). 

Then changing this should be more correct, but we will have to check if it's a bind mount too (we can probably validate it earlier, as otherwise it is not supported).

@thaJeztah What do you think about doing that?